ui-admin/src/forms/designer/split/SplitFormDesigner.tsx

```diff
@@ -16,81 +22,78 @@ export const SplitFormDesigner = ({ content, onChange, currentLanguage, supporte
 }) => {
+  const HIDE_TABLE_OF_CONTENTS_KEY = 'formDesigner.hideTableOfContents'
+
   const [currentPageNo, setCurrentPageNo] = useState(0)
-  const [currentPage, setCurrentPage] = useState(content.pages[currentPageNo])
+  const [hideTableOfContents, setHideTableOfContents] = useState(
+    localStorage.getItem(HIDE_TABLE_OF_CONTENTS_KEY) === 'true')
+  const [searchParams, setSearchParams] = useSearchParams()
+  const selectedElementPath = searchParams.get('selectedElementPath') ?? 'pages'
 
-  const handlePageChange = (direction: 'next' | 'previous') => {
-    if (direction === 'next') {
-      setCurrentPageNo(currentPageNo + 1)
-      setCurrentPage(content.pages[currentPageNo + 1])
-    } else {
-      setCurrentPageNo(currentPageNo - 1)
-      setCurrentPage(content.pages[currentPageNo - 1])
-    }
-    handleScrollToTop()
+  const setSelectedElementPath = (path: string) => {
+    searchParams.set('selectedElementPath', path)
+    setSearchParams(searchParams)
+    scrollToElement(path, setCurrentPageNo)
   }
 
-  const handleScrollToTop = () => {
-    window.scrollTo(0, 0)
-  }
+  useEffect(() => {
+    if (selectedElementPath && selectedElementPath !== 'pages') {
+      scrollToElement(selectedElementPath, setCurrentPageNo)
+    }
+  }, [])
 
-  return <div className="mt-3 container w-100">
-    <div className="d-flex">
-      {renderNewElementButton(content, onChange, -1, currentPageNo, 'question')}
-      {renderNewElementButton(content, onChange, -1, currentPageNo, 'panel')}
-    </div>
-    <>
-      {currentPage && currentPage.elements && currentPage.elements.map((element, elementIndex) => (
-        <div key={elementIndex} className="container">
-          <SplitFormElementDesigner currentPageNo={currentPageNo}
-            elementIndex={elementIndex} editedContent={content}
-            element={content.pages[currentPageNo].elements[elementIndex]}
-            currentLanguage={currentLanguage} supportedLanguages={supportedLanguages}
-            onChange={onChange}/>
-          <div className="d-flex">
-            {renderNewElementButton(content, onChange, elementIndex, currentPageNo, 'question')}
-            {renderNewElementButton(content, onChange, elementIndex, currentPageNo, 'panel')}
+  return <div className="container-fluid overflow-scroll">
+    <div className="row w-100 mx-0">
+      <div style={{ overflowY: 'scroll' }}
+        className={classNames('px-0 border-start border-end bg-white', hideTableOfContents ? 'd-none' : 'col-3')}>
+        { !hideTableOfContents &&
+            <FormTableOfContents
+              formContent={content}
+              selectedElementPath={selectedElementPath}
+              onSelectElement={setSelectedElementPath}
+            />
+        }
+      </div>
+      <div className={classNames('col', hideTableOfContents ? 'col-12' : 'col-9')}>
+        <div className="d-flex justify-content-between border border-top-0 rounded-bottom-3 p-2 bg-light">
```

Comment:
```suggestion
        <div className="d-flex justify-content-between border mt-2 border-0 rounded-3 p-2 bg-light">
```
Another UX nit: Total personal preference, but I think it looks slightly cleaner matching the bottom
<img width="1256" alt="image" src="https://github.com/user-attachments/assets/0f74102d-c6e2-4819-b5cd-3b09c4007304">